core/src/Slay/Prim.hs

```diff
@@ -82,6 +86,18 @@ data PrimCurve g =
       curveDirection :: g Direction
     }
 
+
+-- | circle adt
+data PrimCircle g = Circle
+    { circleColor :: g (Maybe Color)
+    , circleCenterX :: Double
```

Comment:
What do `CenterX` and `CenterY` fields mean? The position of a primitive is determined by the layouting engine, not by the primitive itself.